From: Oever González <notengobattery@gmail.com>
Subject: [PATCH] ipq40xx: improve cpu clock
Date: Sun, 13 Oct 2019 20:42:40 -0600

This patch tries to match the values in the device tree for those found
inside the OEM device tree and kernel source code. This is not
overclocked.

Signed-off-by: Oever González <notengobattery@gmail.com>
---
--- a/arch/arm/boot/dts/qcom-ipq4019.dtsi
+++ b/arch/arm/boot/dts/qcom-ipq4019.dtsi
@@ -61,7 +61,7 @@
 			reg = <0x0>;
 			clocks = <&gcc GCC_APPS_CLK_SRC>;
 			clock-frequency = <0>;
-			clock-latency = <256000>;
+			clock-latency = <100000>;
 			operating-points-v2 = <&cpu0_opp_table>;
 		};
 
@@ -75,7 +75,7 @@
 			reg = <0x1>;
 			clocks = <&gcc GCC_APPS_CLK_SRC>;
 			clock-frequency = <0>;
-			clock-latency = <256000>;
+			clock-latency = <100000>;
 			operating-points-v2 = <&cpu0_opp_table>;
 		};
 
@@ -89,7 +89,7 @@
 			reg = <0x2>;
 			clocks = <&gcc GCC_APPS_CLK_SRC>;
 			clock-frequency = <0>;
-			clock-latency = <256000>;
+			clock-latency = <100000>;
 			operating-points-v2 = <&cpu0_opp_table>;
 		};
 
@@ -103,7 +103,7 @@
 			reg = <0x3>;
 			clocks = <&gcc GCC_APPS_CLK_SRC>;
 			clock-frequency = <0>;
-			clock-latency = <256000>;
+			clock-latency = <100000>;
 			operating-points-v2 = <&cpu0_opp_table>;
 		};
 
@@ -121,22 +121,72 @@
 		opp-48000000 {
 			opp-hz = /bits/ 64 <48000000>;
 			opp-microvolt = <1100000>;
-			clock-latency-ns = <256000>;
+			clock-latency-ns = <100000>;
 		};
 		opp-200000000 {
 			opp-hz = /bits/ 64 <200000000>;
 			opp-microvolt = <1100000>;
-			clock-latency-ns = <256000>;
+			clock-latency-ns = <100000>;
+		};
+		opp-384000000 {
+			opp-hz = /bits/ 64 <384000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <100000>;
+		};
+		opp-413000000 {
+			opp-hz = /bits/ 64 <413000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <100000>;
+		};
+		opp-448000000 {
+			opp-hz = /bits/ 64 <448000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <100000>;
+		};
+		opp-488000000 {
+			opp-hz = /bits/ 64 <488000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <100000>;
 		};
 		opp-500000000 {
 			opp-hz = /bits/ 64 <500000000>;
 			opp-microvolt = <1100000>;
-			clock-latency-ns = <256000>;
+			clock-latency-ns = <100000>;
+		};
+		opp-512000000 {
+			opp-hz = /bits/ 64 <512000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <100000>;
+		};
+		opp-537000000 {
+			opp-hz = /bits/ 64 <537000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <100000>;
+		};
+		opp-565000000 {
+			opp-hz = /bits/ 64 <565000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <100000>;
 		};
+		opp-597000000 {
+			opp-hz = /bits/ 64 <597000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <100000>;
+		};
+		opp-632000000 {
+			opp-hz = /bits/ 64 <632000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <100000>;
+		};
+		opp-672000000 {
+			opp-hz = /bits/ 64 <672000000>;
+			opp-microvolt = <1100000>;
+			clock-latency-ns = <100000>;
+		};
 		opp-716000000 {
 			opp-hz = /bits/ 64 <716000000>;
 			opp-microvolt = <1100000>;
-			clock-latency-ns = <256000>;
+			clock-latency-ns = <100000>;
- 		};
+		};
 	};
 
